### PR TITLE
feat(examples/sdxl): upgrade sdxl to ms2.3(jit_level/dynamic shape)

### PR DESCRIPTION
**Pull request by wcrzlh**
# What does this PR do?

Adds # (feature)
- [x] feat(examples/sdxl): upgrade sdxl to ms2.3(jit_level/dynamic shape-vanilla-lora-dreambooth)

Fixes # (issue)
- [x] unlimit the number restriction of aspect ratio
- [x] fix sdxl DreamBooth memory bug

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [ ] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

**Review by CaitinZhao — APPROVED** [review]

**Review by zhanghuiyao — APPROVED** [review]

**Review by zhanghuiyao — APPROVED** [review]

File: examples/stable_diffusion_xl/gm/helpers.py [hunk 1]
```diff
@@ -111,6 +111,19 @@ def set_default(args):
...
+            ms.set_context(jit_config={"jit_level": args.jit_level})
+            print(f"set jit_level: {args.jit_level}.")
+        else:
+            print(
+                f"WARNING: Unsupport jit_level: {args.jit_level}. The framework automatically selects the execution method"
+            )
+    except Exception:
+        print(
+            "WARNING: The current jit_level is not suitable because current MindSpore version or mode does not match,"
+            "please ensure the MindSpore version >= ms2.3_0615, and use GRAPH_MODE."
```
Comment (Songyuanwei) [inline examples/stable_diffusion_xl/gm/helpers.py:125]: version >= ms2.3.0, 不需要mindspore具体日期

File: examples/stable_diffusion_xl/train_textual_inversion.py [hunk 2]
```diff
@@ -67,6 +67,16 @@ def get_parser_train():
     parser.add_argument(
         "--ms_mode", type=int, default=0, help="Running in GRAPH_MODE(0) or PYNATIVE_MODE(1) (default=1)"
     )
+    parser.add_argument(
+        "--jit_level",
+        default="O2",
+        type=str,
+        choices=["O0", "O1", "O2"],
+        help="Used to control the compilation optimization level. Supports [“O0”, “O1”, “O2”]."
```
Comment (Songyuanwei) [inline examples/stable_diffusion_xl/train_textual_inversion.py:75]: 这个是不是中文双引号？应该换成英文的。

File: examples/stable_diffusion_xl/gm/helpers.py (hunk 1 above)
Comment (wcrzlh) [reply examples/stable_diffusion_xl/gm/helpers.py]: fixed

File: examples/stable_diffusion_xl/train_textual_inversion.py (hunk 2 above)
Comment (wcrzlh) [reply examples/stable_diffusion_xl/train_textual_inversion.py]: fixed

**Review by CaitinZhao — APPROVED** [review]

File: examples/stable_diffusion_xl/gm/modules/diffusionmodules/model.py [hunk 3]
```diff
@@ -49,7 +48,12 @@ def __init__(self, in_channels, with_conv):
 
     def construct(self, x):
         if self.with_conv:
-            x = mnp.pad(x, ((0, 0), (0, 0), (0, 1), (0, 1)))
+            b, c, h, w = x.shape[0], x.shape[1], x.shape[2], x.shape[-1]
+            concat_h = ops.ones((b, c, h, 1), x.dtype)
+            concat_w = ops.ones((b, c, 1, w + 1), x.dtype)
+
+            x = ops.Concat(3)([x, concat_h])
```
Comment (SamitHuang) [inline examples/stable_diffusion_xl/gm/modules/diffusionmodules/model.py:55]: use ops.concat

Comment (wcrzlh) [reply examples/stable_diffusion_xl/gm/modules/diffusionmodules/model.py]: modified

**Review by vigo999 — APPROVED** [review]

**Pull request merged**